### PR TITLE
Orleans observability: logging, metrics and distributed tracing

### LINKED ISSUE #33745: Not for latest Orleans version
Seems like this is only applicable to the older Orleans , not v7. It isn't always clear in the docs and that makes it hard to navigate,

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 6e987171-00a7-5cd4-8b0d-674e4e5415fb
* Version Independent ID: b75215a0-bb33-efe0-5eb4-8b0e1b35fa41
* Content: [Runtime monitoring and logs](https://learn.microsoft.com/en-us/dotnet/orleans/host/monitoring/)
* Content Source: [docs/orleans/host/monitoring/index.md](https://github.com/dotnet/docs/blob/main/docs/orleans/host/monitoring/index.md)
* Product: **dotnet-orleans**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

---
[Associated WorkItem - 62388](https://dev.azure.com/msft-skilling/Content/_workitems/edit/62388)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

Orleans observability: logging, metrics, and distributed tracing

Fix #33745, @bradygaster @ReubenBond, also notice that the name changed from "Runtime monitoring" to "Orleans observability".

Previews:

- [Orleans observability](https://review.learn.microsoft.com/en-us/dotnet/orleans/host/monitoring/?branch=pr-en-us-34229&pivots=orleans-7-0)
- [Orleans hub](https://review.learn.microsoft.com/en-us/dotnet/orleans/?branch=pr-en-us-34229)

### CODE REVIEW

**Review by ReubenBond — APPROVED** [review]

File: docs/orleans/host/monitoring/index.md [hunk 1]
```diff
@@ -1,14 +1,158 @@
...
+
+Orleans leverages [Microsoft.Extensions.Logging](https://www.nuget.org/packages/Microsoft.Extensions.Logging) for all silo and client logs. This means that you can use any logging provider that is compatible with `Microsoft.Extensions.Logging`. Your app code would rely on [dependency injection](../../../core/extensions/dependency-injection.md) to get an instance of <xref:Microsoft.Extensions.Logging.ILogger%601> and use it to log messages. For more information, see [Logging in .NET](../../../core/extensions/logging.md).
+
+<!-- markdownlint-disable MD044 -->
+:::zone target="docs" pivot="orleans-7-0"
+<!-- markdownlint-enable MD044 -->
+
+## Metrics
+
+Metrics are numerical measurements reported over time, most often used to monitor the health of an application and generate alerts. For more information, see [Metrics in .NET](../../../core/diagnostics/metrics.md). Orleans uses the [System.Diagnostics.Metrics](../../../core/diagnostics/compare-metric-apis.md#systemdiagnosticsmetrics) APIs to collect metrics. The metrics are exposed to the [OpenTelemetry](https://opentelemetry.io) project, which exports the metrics to various monitoring systems.
```
Comment (gewarren) [inline docs/orleans/host/monitoring/index.md:22]:
```suggestion
Metrics are numerical measurements reported over time. They're most often used to monitor the health of an application and generate alerts. For more information, see [Metrics in .NET](../../../core/diagnostics/metrics.md). Orleans uses the [System.Diagnostics.Metrics](../../../core/diagnostics/compare-metric-apis.md#systemdiagnosticsmetrics) APIs to collect metrics. The metrics are exposed to the [OpenTelemetry](https://opentelemetry.io) project, which exports the metrics to various monitoring systems.
```

File: docs/orleans/host/monitoring/index.md [hunk 2]
```diff
@@ -1,14 +1,158 @@
...
+
+To monitor your app without making any code changes at all, you can use the `dotnet counters` .NET diagnostic tool. To monitor Orleans <xref:System.Diagnostics.ActivitySource> counters, given your desired `{ProcessName}` to monitor, use the `dotnet counters monitor` command as shown:
```
Comment (gewarren) [inline docs/orleans/host/monitoring/index.md:24]:
```suggestion
To monitor your app without making any code changes at all, you can use the `dotnet counters` .NET diagnostic tool. To monitor Orleans <xref:System.Diagnostics.ActivitySource> counters, given your desired `<ProcessName>` to monitor, use the `dotnet counters monitor` command as shown:
```

Comment (gewarren) [reply docs/orleans/host/monitoring/index.md:24]: Angle bracket guidance is from https://review.learn.microsoft.com/en-us/help/contribute/text-formatting-guidelines?branch=main#placeholders.

File: docs/orleans/host/monitoring/index.md [hunk 3]
```diff
@@ -1,14 +1,158 @@
...
+
+```dotnetcli
+dotnet counters monitor -n {ProcessName} --counters Microsoft.Orleans
```
Comment (gewarren) [inline docs/orleans/host/monitoring/index.md:27]:
```suggestion
dotnet counters monitor -n <ProcessName> --counters Microsoft.Orleans
```

File: docs/orleans/host/monitoring/index.md [hunk 4]
```diff
@@ -1,14 +1,158 @@
...
+    .WithMetrics(metrics =>
+    {
+        metrics
+            .AddPrometheusExporter()
+            .AddMeter("Microsoft.Orleans");
+    });
+```
+
+> [!IMPORTANT]
+> Both the [OpenTelemetry.Exporter.Prometheus](https://www.nuget.org/packages/OpenTelemetry.Exporter.Prometheus) and [OpenTelemetry.Exporter.Prometheus.AspNetCore](https://www.nuget.org/packages/OpenTelemetry.Exporter.Prometheus.AspNetCore) NuGet packages are currently in preview as a release candidates. They are not recommended for production use.
```
Comment (gewarren) [inline docs/orleans/host/monitoring/index.md:99]:
```suggestion
> Both the [OpenTelemetry.Exporter.Prometheus](https://www.nuget.org/packages/OpenTelemetry.Exporter.Prometheus) and [OpenTelemetry.Exporter.Prometheus.AspNetCore](https://www.nuget.org/packages/OpenTelemetry.Exporter.Prometheus.AspNetCore) NuGet packages are currently in preview as release candidates. They're not recommended for production use.
```

File: docs/orleans/host/monitoring/index.md [hunk 5]
```diff
@@ -1,14 +1,158 @@
...
+## Distributed tracing
+
+Distributed tracing is a set of tools and practices to monitor and troubleshoot distributed applications. Distributed tracing is a key component of observability, and it's a critical tool for developers to understand the behavior of their apps. Orleans also supports distributed tracing with [OpenTelemetry](https://opentelemetry.io).
+
+Regardless of the distributed tracing exporter you choose, you'll call:
+
+- <xref:Orleans.Hosting.CoreHostingExtensions.AddActivityPropagation(Orleans.Hosting.ISiloBuilder)>: To enable distributed tracing for the silo.
+- <xref:Orleans.Hosting.ClientBuilderExtensions.AddActivityPropagation(Orleans.Hosting.IClientBuilder)>: To enable distributed tracing for the client.
+
+Relying again on the [Orleans GPS Tracker sample app](/samples/dotnet/samples/orleans-gps-device-tracker-sample), you can use the [Zipkin](https://zipkin.io) distributed tracing system to monitor the app. To use OpenTelemetry and Zipkin with Orleans, call the following `IServiceCollection` extension method:
```
Comment (gewarren) [inline docs/orleans/host/monitoring/index.md:121]: This sentence is not clear. What relies on the sample app, Zipkin or the code in the snippet?

File: docs/orleans/toc.yml [hunk 6]
```diff
@@ -152,10 +152,12 @@ items:
             href: host/configuration-guide/startup-tasks.md
           - name: Orleans graceful shut down
             href: host/configuration-guide/shutting-down-orleans.md
-      - name: Monitoring
+      - name: Observability
+        displayName: observability,logging,monitoring,distributed tracing,metrics,telemetry
```
Comment (gewarren) [inline docs/orleans/toc.yml:156]: This won't have any effect since there's no file at this node.

```suggestion
```

File: docs/orleans/toc.yml [hunk 7]
```diff
@@ -152,10 +152,12 @@ items:
...
         items:
-          - name: Runtime monitoring
+          - name: Overview
             href: host/monitoring/index.md
+            displayName: zipkin,jaeger,open telemetry,opentelemetry,meter,counter,activity source
```
Comment (gewarren) [inline docs/orleans/toc.yml:160]:
```suggestion
            displayName: zipkin,jaeger,open telemetry,opentelemetry,meter,counter,activity source,observability,monitoring,logging,metrics,telemetry,distributed tracing
```

**Review by gewarren — APPROVED** [review]

File: docs/orleans/host/monitoring/index.md (hunk 5 above)
Comment (IEvangelist) [inline docs/orleans/host/monitoring/index.md:121]:
```suggestion
Referring back to the [Orleans GPS Tracker sample app](/samples/dotnet/samples/orleans-gps-device-tracker-sample), you can use the [Zipkin](https://zipkin.io) distributed tracing system to monitor the app by updating the _Program.cs_. To use OpenTelemetry and Zipkin with Orleans, call the following `IServiceCollection` extension method:
```

**Pull request merged**